### PR TITLE
Propagate error from API response to the MarathonException

### LINKED ISSUE #2: Error details from the response not propagated through client
How to reproduce:
- use client to create invalid request on Marathon API (e.g. trigger validation error)
- MarathonException is thrown, no details from the response are propagated back to the user

It would be nice to see contents of the Error response, at least a message

### PR DESCRIPTION
**Pull request by alenkacz**
fixes #2

### CODE REVIEW

**Review by kensipe — APPROVED** [review]
I love the additions...   Our forward test direction is using Spock which I think would clean up the tests you created.    There is no reason to block this tho...

**Comment by kensipe** [comment]
@alenkacz let me know if you merge as is... I will add a todo to convert the test.

**Comment by kensipe** [comment]
@colin-msphere could we get your review as well?

File: src/main/java/mesosphere/marathon/client/MarathonClient.java [hunk 1]
```diff
@@ -27,6 +30,14 @@ public void apply(RequestTemplate template) {
 	static class MarathonErrorDecoder implements ErrorDecoder {
 		@Override
 		public Exception decode(String methodKey, Response response) {
+			if (response.status() >= 400 && response.status() < 500 ) {
```
Comment (colin-msphere) [inline src/main/java/mesosphere/marathon/client/MarathonClient.java:33]: Why was this range chosen?

**Review by colin-msphere — COMMENTED** [review]
I agree with Ken and would like to see the client test in the existing spock test.

I had one question on status codes.

**Comment by ghost** [comment]
@colin-msphere  Do you think this range is not appropriated

**Comment by kensipe** [comment]
ok... this is useful... my only real issue is the choice in testing tools... we want to move towards spock... but we have other tests that need to be converted as well.  Merging..

**Pull request merged**